src/engraving/libmscore/stemslash.cpp

```diff
@@ -67,22 +68,62 @@ void StemSlash::setLine(const LineF& l)
 
 void StemSlash::layout()
 {
-    Stem* stem = chord()->stem();
-    double h2;
-    double _spatium = spatium();
-    double l = chord()->up() ? _spatium : -_spatium;
-    PointF p(stem->flagPosition());
-    double x = p.x() + _spatium * .1;
-    double y = p.y();
-
-    if (chord()->beam()) {
-        y += l * .3;
-        h2 = l * .8;
+    if (!chord() || !chord()->stem()) {
+        return;
+    }
+    Chord* c = chord();
+    Stem* stem = c->stem();
+    Hook* hook = c->hook();
+    Beam* beam = c->beam();
+
+    static constexpr double heightReduction = 0.66;
+    static constexpr double angleIncrease = 1.2;
+    static constexpr double lengthIncrease = 1.1;
+
+    double up = c->up() ? -1 : 1;
+    double stemTipY = c->up() ? stem->bbox().translated(stem->pos()).top() : stem->bbox().translated(stem->pos()).bottom();
+    double leftHang = score()->noteHeadWidth() * score()->styleD(Sid::graceNoteMag) / 2;
+    double angle = score()->styleD(Sid::stemSlashAngle) * M_PI / 180; // converting to radians
+    bool straight = score()->styleB(Sid::useStraightNoteFlags);
+    double graceNoteMag = score()->styleD(Sid::graceNoteMag);
+
+    double startX = stem->bbox().translated(stem->pos()).right() - leftHang;
+
+    double startY;
+    if (straight || beam) {
+        startY = stemTipY - up * graceNoteMag * score()->styleMM(Sid::stemSlashHeight) * heightReduction;
     } else {
-        y += l * 1.2;
-        h2 = l * .4;
+        startY = stemTipY - up * graceNoteMag * score()->styleMM(Sid::stemSlashHeight);
     }
-    double w  = chord()->upNote()->bboxRightPos() * .7;
-    setLine(LineF(PointF(x + w, y - h2), PointF(x - w, y + h2)));
+
+    double endX = 0;
+    double endY = 0;
+
+    if (hook) {
+        auto musicFont = score()->styleSt(Sid::MusicalSymbolFont);
+        // HACK: adjust slash angle for fonts with "fat" hooks. In future, we must use smufl cutOut
+        if (c->ticks() < Fraction(1, 8) && !straight
```

Comment:
Could we also use `c->beams() >= 2` or something in that direction here?

Reply:
Ha! Thanks, didn't know we had that